core/contextmenu.js

```diff
@@ -139,6 +150,20 @@ Blockly.ContextMenu.position_ = function(menu, e, rtl) {
     Blockly.utils.uiMenu.adjustBBoxesForRTL(viewportBBox, anchorBBox, menuSize);
   }
 
+  // Closure Library menu code includes a fix for a bug where a menu option
+  // will respond to the event which was responsible for showing the menu in
+  // the first place.  This fix compares the coordinates of the first mouseup
+  // event with those of the event which created it (made the menu visible).
+  // We can only take advantage of the fix if we specifically provide the
+  // mouse event to the menu, even if we don't otherwise have any reason to
+  // call setVisible (as the menu defaults to visible anyway).
+  // See issue #2085.
+  menu.setVisible(
+    true, // this menu is visible (this is the default anyway)
+    true, // "force" the value and don't emit SHOW & AFTER_SHOW events (avoid side effects)
```

Comment:
Technically this line ("force") could be `false` since the menu is visible by default anyway, and so its value doesn't change. It would avoid [some slight overhead](https://github.com/google/closure-library/blob/8f2a4aaa33b31bb8aa07dccb7e33ab67dc756b8b/closure/goog/ui/container.js#L1116-L1123) of re-applying certain DOM values, and would be less misleading without any `SHOW`/`AFTER_SHOW` side effects—but it necessitates familiarity/assumption that the menu starts out visible. (Obviously we can take this for granted, or else menu rendering as-is wouldn't work in the first place.)